service/src/main/kotlin/fi/espoo/evaka/shared/security/actionrule/HasGlobalRole.kt

```diff
@@ -162,4 +165,26 @@ WHERE pd.provider_type = 'PRIVATE_SERVICE_VOUCHER'
                     .trimIndent()
             )
         }
+
+    fun andUnitProviderAndCareTypeEquals(
+        providerTypes: Set<ProviderType>,
+        careTypes: Set<CareType>
+    ) =
+        rule<DaycareId> { _, _ ->
+            sql(
+                """
+SELECT id
+FROM daycare
+WHERE provider_type = ANY(${bind(providerTypes)})
+${
+    if (careTypes.isNotEmpty()) "AND (${careTypes.mapIndexedNotNull{ i, value -> 
+            if (i == 0) "${"'$value'"} = ANY(daycare.type)"
+            else " OR ${"'$value'"} = ANY(daycare.type)"
+        }.joinToString(" ")})"
```

Comment:
I think this could be much simpler. Something like
`careTypes.map { "$it" = ANY(daycare.type)" }.joinToString(" OR ")`

Reply:
actually concatenating OR statements isnt even necessary. Isn't this what we want:
```
anyarray && anyarray → boolean

Do the arrays overlap, that is, have any elements in common?
```
So simply `${bind(careTypes)} && daycare.type`